### PR TITLE
#101006 create zaken api plugin action start hersteltermijn

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/ZakenApiPlugin.kt [hunk 1]
```diff
@@ -331,6 +333,44 @@ class ZakenApiPlugin(
         )
     }
 
+    @PluginAction(
+        key = "start-hersteltermijn",
+        title = "Start hersteltermijn",
+        description = "Start the recovery period for a case",
+        activityTypes = [SERVICE_TASK_START, USER_TASK_CREATE]
+    )
+    @Transactional
```
Comment (theo-ritense) [inline zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/ZakenApiPlugin.kt]: A plugin is not managed by spring, so I doubt transactions work like this.

File: zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/ZakenApiPlugin.kt [hunk 2]
```diff
@@ -331,6 +333,44 @@ class ZakenApiPlugin(
...
+            val uiterlijkeEinddatumAfdoening = zaak.uiterlijkeEinddatumAfdoening
+                ?: calculateUiterlijkeEinddatumAfdoening(zaak.zaaktype, zaak.startdatum)
+            require(uiterlijkeEinddatumAfdoening != null) { "No 'uiterlijkeEinddatumAfdoening' available for zaak '$zaakUrl' " }
+
+            client.patchZaak(
+                authenticationPluginConfiguration, url, PatchZaakRequest(
+                    uiterlijkeEinddatumAfdoening = uiterlijkeEinddatumAfdoening.plusDays(maxDurationInDays.toLong())
+                )
+            )
+            zaakHersteltermijnRepository.save(hersteltermijn)
```
Comment (theo-ritense) [inline zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/ZakenApiPlugin.kt]: Please use a spring-managed service instead so you can do transactional stuff.

File: zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/ZakenApiPlugin.kt [hunk 3]
```diff
@@ -61,12 +66,13 @@ import java.util.UUID
...
     description = "Connects to the Zaken API"
 )
-class ZakenApiPlugin(
+open class ZakenApiPlugin(
     private val client: ZakenApiClient,
     private val zaakUrlProvider: ZaakUrlProvider,
     private val storageService: TemporaryResourceStorageService,
     private val zaakInstanceLinkRepository: ZaakInstanceLinkRepository,
     private val pluginService: PluginService,
+    private val zaakHersteltermijnRepository: ZaakHersteltermijnRepository,
```
Comment (theo-ritense) [inline zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/ZakenApiPlugin.kt:76]: Please use a spring-managed service instead so you can do transactional stuff.

File: zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/ZakenApiPlugin.kt (hunk 2 above)
Comment (Klaas-Ritense) [reply zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/ZakenApiPlugin.kt]: I've changed it to use a platformTransactionManager

**Review by marijnritense — APPROVED** [review]

**Pull request merged**